sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/implementation/VirtualMachineUpdateTests.java

```diff
@@ -97,14 +99,20 @@ public void testVirtualMachineUpdate() {
         Assertions.assertEquals("value1", vm.tags().get("key1"));
 
         // modify msi
-        vm.update()
+        String identityName1 = generateRandomResourceName("msi-id", 15);
+        Creatable<Identity> identityToCreate = msiManager.identities()
+            .define(identityName1)
+            .withRegion(Region.US_WEST3)
+            .withExistingResourceGroup(rgName);
+        vmUpdate = vm.update()
             .withSystemAssignedManagedServiceIdentity()
             .withSystemAssignedIdentityBasedAccessToCurrentResourceGroup(BuiltInRole.CONTRIBUTOR)
-            .apply();
+            .withNewUserAssignedManagedServiceIdentity(identityToCreate);
         Assertions.assertTrue(this.isVirtualMachineModifiedDuringUpdate(vm));
+        vmUpdate.apply();
 
         // verify msi
-        Assertions.assertEquals(ResourceIdentityType.SYSTEM_ASSIGNED, vm.managedServiceIdentityType());
+        Assertions.assertEquals(ResourceIdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED, vm.managedServiceIdentityType());
         Assertions.assertNotNull(vm.systemAssignedManagedServiceIdentityPrincipalId());
```

Comment:
Curious how did you find the test case issue, as I guess there won't be failure in test even VM is not updated?

Reply:
The failure is at
```
        Assertions.assertTrue(this.isVirtualMachineModifiedDuringUpdate(vm));
```

Reply:
ok, I see, thanks.